index.js

```diff
@@ -192,6 +193,93 @@ module.exports = class HyperDHT extends DHT {
     }
   }
 
+  async immutableGet (key) {
+    if (Buffer.isBuffer(key) === false) throw new Error('key must be a buffer')
+    const query = this.query(key, 'immutable_get', null, { map: mapImmutable })
+    const check = Buffer.allocUnsafe(32)
+    for await (const node of query) {
+      const { value } = node
+      sodium.crypto_generichash(check, value)
+      if (check.equals(key)) return node
+    }
+    throw Error('not found')
+  }
+
+  async immutablePut (value) {
+    if (Buffer.isBuffer(value) === false) throw new Error('value must be a buffer')
+    if (value.length > PUT_VALUE_MAX_SIZE) {
+      throw new Error(`Value size must be <= ${PUT_VALUE_MAX_SIZE}`)
+    }
+    const key = Buffer.allocUnsafe(32)
+    sodium.crypto_generichash(key, value)
+    const query = this.query(key, 'immutable_get', null, {
+      map: mapImmutable,
+      commit (node, dht) {
+        return dht.request(key, 'immutable_put', value, node.from, { map: mapImmutable, token: node.token })
+      }
+    })
+    await query.finished()
+    return { key }
+  }
+
+  async mutableGet (key, { salt = null, seq = 0, latest = true } = {}) {
+    if (Buffer.isBuffer(key) === false) throw new Error('key must be a buffer')
+    if (typeof seq !== 'number') throw new Error('seq should be a number')
+    if (salt) {
+      if (Buffer.isBuffer(salt) === false) throw new Error('salt must be a buffer')
+      if (salt.length > 64) throw new Error('salt size must be no greater than 64 bytes')
+    }
+
+    const request = cenc.encode(messages.mutableReq, { salt, seq })
+
+    const query = this.query(key, 'mutable_get', request, { map: mapMutable })
+    const userSeq = seq
+    let topSeq = seq
```

Comment:
would it make sense to rename seq in the opts to `gt` and `gte` ? (that's how our db abstractions name the limits)

Reply:
ya lets keep at seq